arango/collection.py

```diff
@@ -2006,21 +2016,25 @@ def import_bulk(
         if sync is not None:
             params["waitForSync"] = sync
 
-        request = Request(
-            method="post",
-            endpoint="/_api/import",
-            data=documents,
-            params=params,
-            write=self.name,
-        )
-
         def response_handler(resp: Response) -> Json:
             if resp.is_success:
                 result: Json = resp.body
                 return result
             raise DocumentInsertError(resp, request)
 
-        return self._execute(request, response_handler)
+        results = []
+        for batch in get_batches(documents, batch_size or len(documents)):
+            request = Request(
+                method="post",
+                endpoint="/_api/import",
+                data=batch,
+                params=params,
+                write=self.name,
+            )
+
+            results.append(self._execute(request, response_handler))
+
+        return results[0] if batch_size is None else results
```

Comment:
See https://github.com/ArangoDB-Community/python-arango/pull/207#discussion_r904289441.

Reply:
Addressed in [c65af2a](https://github.com/ArangoDB-Community/python-arango/pull/207/commits/c65af2add238347c12d990a70bb7fb733a0a0ed7)